modules/tinymce/src/plugins/image/test/ts/browser/ContextMenuTest.ts

```diff
@@ -36,6 +50,25 @@ describe('browser.tinymce.plugins.image.ContextMenuTest', () => {
     TinyAssertions.assertSelection(editor, [], 0, [], 1);
   });
 
+  it('TINY-9491: Opening context menus on a selected figure', async () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<div contenteditable="false">' +
+        '<h2 contenteditable="true">Table</h2>' +
+        '<ul>' +
+          '<li>' +
+            '<a href="#mcetoc_1gmi3gu5b11" data-mce-href="#mcetoc_1gmi3gu5b11">point </a>' +
+          '</li>' +
+        '</ul>' +
+      '</div>' +
+      '<h2 id="mcetoc_1gmi3gu5b12">point</h2>',
+      { format: 'raw' }
+    );
+    Mouse.contextMenuOn(TinyDom.body(editor), 'a');
+    await pOpenContextMenu(editor, 'a');
+    UiFinder.notExists(SugarShadowDom.getContentContainer(SugarShadowDom.getRootNode(TinyDom.targetElement(editor))), 'div[title="Link..."');
```

Comment:
Curious why what all the shadow dom magic needed here? From what I can see the editor is not rendered in a shadow dom. Couldn't you just look for the selector on SugarBody.body() and maybe make it more specific the selector if possible to include that the title is not part of some other structure since in theory that div could be in the toolbar.

Reply:
I don't know, honestly. When I went going for stuff I went and fetched the function calls we have in the functions ( such as the opening of the context toolbar ) and used those. Since those do work I thought it'd be best to just use them.